django/db/migrations/state.py

```diff
@@ -95,10 +95,13 @@ def __init__(self, models=None, real_apps=None):
         self.is_delayed = False
         # {remote_model_key: {model_key: [(field_name, field)]}}
         self.relations = None
+        self.resolve_fields_and_relations()
 
     def add_model(self, model_state):
         app_label, model_name = model_state.app_label, model_state.name_lower
         self.models[(app_label, model_name)] = model_state
+        concretes, _ = self._get_concrete_models_mapping_and_proxy_models()
+        self.populate_relation_from_model_state(model_state, (app_label, model_name), concretes)
```

Comment:
This should only be performed if the `relations` registry is already computed; `if 'relations' in self.__dict__`